plugin/pkg/scheduler/core/generic_scheduler.go

```diff
@@ -423,11 +487,322 @@ func EqualPriorityMap(_ *v1.Pod, _ interface{}, nodeInfo *schedulercache.NodeInf
 	}, nil
 }
 
+// pickOneNodeForPreemption chooses one node among the given nodes. It assumes
+// pods in each map entry are ordered by decreasing priority.
+// It picks a node based on the following criteria:
+// 1. A node with minimum highest priority victim is picked.
+// 2. Ties are broken by sum of priorities of all victims.
+// 3. If there are still ties, node with the minimum number of victims is picked.
+// 4. If there are still ties, the first such node in the map is picked (sort of randomly).
+func pickOneNodeForPreemption(nodesToPods map[*v1.Node][]*v1.Pod) *v1.Node {
+	type nodeScore struct {
+		node            *v1.Node
+		highestPriority int32
+		sumPriorities   int64
+		numPods         int
+	}
+	if len(nodesToPods) == 0 {
+		return nil
+	}
+	minHighestPriority := int32(math.MaxInt32)
+	nodeScores := []*nodeScore{}
+	for node, pods := range nodesToPods {
+		if len(pods) == 0 {
+			// We found a node that doesn't need any preemption. Return it!
+			// This should happen rarely when one or more pods are terminated between
+			// the time that scheduler tries to schedule the pod and the time that
+			// preemption logic tries to find nodes for preemption.
+			return node
+		}
+		// highestPodPriority is the highest priority among the victims on this node.
+		highestPodPriority := util.GetPodPriority(pods[0])
+		if highestPodPriority < minHighestPriority {
+			minHighestPriority = highestPodPriority
+		}
+		if highestPodPriority == minHighestPriority {
+			nodeScores = append(nodeScores, &nodeScore{node: node, highestPriority: highestPodPriority, numPods: len(pods)})
+		}
+	}
+	// Find the nodes with minimum highest priority victim.
+	minSumPriorities := int64(math.MaxInt64)
+	lowestHighPriorityNodes := []*nodeScore{}
+	for _, nodeScore := range nodeScores {
+		if nodeScore.highestPriority == minHighestPriority {
+			lowestHighPriorityNodes = append(lowestHighPriorityNodes, nodeScore)
+			var sumPriorities int64
+			for _, pod := range nodesToPods[nodeScore.node] {
+				// We add MaxInt32+1 to all priorities to make all of them >= 0. This is
+				// needed so that a node with a few pods with negative priority is not
+				// picked over a node with a smaller number of pods with the same negative
+				// priority (and similar scenarios).
+				sumPriorities += int64(util.GetPodPriority(pod)) + int64(math.MaxInt32+1)
+			}
+			if sumPriorities < minSumPriorities {
+				minSumPriorities = sumPriorities
+			}
+			nodeScore.sumPriorities = sumPriorities
+		}
+	}
+	if len(lowestHighPriorityNodes) == 1 {
+		return lowestHighPriorityNodes[0].node
+	}
+	// There are multiple nodes with the same minimum highest priority victim.
+	// Choose the one(s) with lowest sum of priorities.
+	minNumPods := math.MaxInt32
+	lowestSumPriorityNodes := []*nodeScore{}
+	for _, nodeScore := range lowestHighPriorityNodes {
+		if nodeScore.sumPriorities == minSumPriorities {
+			lowestSumPriorityNodes = append(lowestSumPriorityNodes, nodeScore)
```

Comment:
The same here - we can do two phases of filtering here. We can store only those with minSumPriorities and from these only those that have minimal number of pods (the same trick I mentioned above).

Reply:
Done.